pygments/lexers/soong.py

```diff
@@ -0,0 +1,77 @@
+"""
+    pygments.lexers.soong
+    ~~~~~~~~~~~~~~~~~~~~~
+
+    Lexers for Soong (Android.bp Blueprint) files.
+
+    :copyright: Copyright 2006-2024 by the Pygments team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+from pygments.lexer import RegexLexer, bygroups, include
+from pygments.token import Comment, Name, Number, Operator, Punctuation, \
+        String, Whitespace
+
+__all__ = ['SoongLexer']
+
+class SoongLexer(RegexLexer):
+    name = 'Soong'
+    version_added = '2.18'
+    url = 'https://source.android.com/docs/setup/reference/androidbp'
+    aliases = ['androidbp', 'bp', 'soong']
+    filenames = [
+        'Android.bp',
+    ]
+
+    tokens = {
+        'root': [
+            # A variable assignment
+            (r'(\w*)(\s*)(\+?=)(\s*)',
+             bygroups(Name.Variable, Whitespace, Operator, Whitespace),
+             'expr'),
+
+            # A top-level module
+            (r'(\w*)(\s*)(\{)',
+             bygroups(Name.Function, Whitespace, Punctuation),
+             'in-rule'),
+
+            # Everything else
+            include('comments'),
+            (r'\s+', Whitespace),  # newlines okay
+        ],
+        'in-list': [
+            include('expr'),
+            include('comments'),
+            (r'\s+', Whitespace),  # newlines okay in a list
+            (r',', Punctuation),
+            (r'\]', Punctuation, '#pop'),
+        ],
+        'in-map': [
+            # A map key
+            (r'(\w+)(:)(\s*)', bygroups(Name, Punctuation, Whitespace)),
+
+            include('expr'),
+            include('comments'),
+            (r'\s+', Whitespace),  # newlines okay in a map
+            (r',', Punctuation),
+            (r'\}', Punctuation, '#pop'),
+        ],
+        'in-rule': [
+            # Just re-use map syntax
+            include('in-map'),
+        ],
+        'comments': [
+            (r'//.*?$', Comment.Single),
```

Comment:
Unnecessary:

```suggestion
            (r'//.*?', Comment.Single),
```

Reply:
Removing `$` seems to break everything.

I think the reason is because `.*?` is non-greedy. If you take away the subsequent character, it will match nothing:

https://docs.python.org/3/howto/regex.html#greedy-versus-non-greedy

```Python Console
>>> re.match(r'//.*?', '// foo!')
<re.Match object; span=(0, 2), match='//'>
>>> 
>>> re.match(r'//.*?!', '// foo!')
<re.Match object; span=(0, 7), match='// foo!'>
```

---

I took this from one of the other lexers. If you search, there are others exactly like this:
```console
$ git grep "\/\/\.\*\?\$', Comment"
```

Reply:
Oh in fact, this comes directly from [the docs](https://pygments.org/docs/lexerdevelopment/#changing-states).

Reply:
Oh, sorry, of course I meant to also remove the `?` at the end (and I've just made that change in the docs).

Reply:
Change applied.

Reply:
Got it. I made the change with no impact to the output.